### PR TITLE
feat: improve typing for `to_pandas` / `to_numpy` / `to_arrow`

### PR DESCRIPTION
**Pull request by MarcoGorelli**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/dataframe.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
     from pathlib import Path
 
     import numpy as np
+    import pandas as pd
+    import pyarrow as pa
```
Comment (FBruzzesi) [inline narwhals/dataframe.py:25]: How would this work if I am in a dev env without one of these libraries?

Comment (MarcoGorelli) [reply narwhals/dataframe.py:25]: IIUC it would just be reported as `Any`

File: narwhals/_arrow/series.py [hunk 2]
```diff
@@ -393,7 +392,8 @@ def is_in(self, other: Any) -> Self:
         return self._from_native_series(pc.is_in(ser, value_set=value_set))
 
     def arg_true(self) -> Self:
-        np = get_numpy()
+        import numpy as np  # ignore-banned-import
```
Comment (MarcoGorelli) [inline narwhals/_arrow/series.py:395]: currently, `import pyarrow` also imports NumPy

but I don't know if that's something we can rely on never changing in the future

so I think we should explicitly import numpy here

i'll add some tests regarding imports, that's probably better than relying on these `get_` functions

**Comment by MarcoGorelli** [comment]
merging for now so I can use `# ignore-banned-import` in the other pr

File: narwhals/dataframe.py (hunk 1 above)
Comment (MarcoGorelli) [reply narwhals/dataframe.py:25]: @FBruzzesi , just to give a quick demo with the latest Narwhals installed in a new venv, in which I have Polars but not Modin:
```
import narwhals as nw
from narwhals.dependencies import is_modin_dataframe, is_polars_dataframe
from narwhals.typing import IntoDataFrame

def func(df: IntoDataFrame) -> None:
    if is_modin_dataframe(df):
        reveal_type(df)
    elif is_polars_dataframe(df):
        reveal_type(df)
```
outputs
```
$ mypy t.py 
t.py:7: note: Revealed type is "Any"
t.py:9: note: Revealed type is "polars.dataframe.frame.DataFrame"
```

**Pull request merged**